lib/plivo/resources/messages.rb

```diff
@@ -61,60 +61,129 @@ def get(message_uuid)
       # @option options [String] :method The method used to call the url. Defaults to POST.
       # @option options [String] :log If set to false, the content of this message will not be logged on the Plivo infrastructure and the dst value will be masked (e.g., 141XXXXX528). Default is set to true.
       # @option options [String] :trackable set to false
-      def create(src, dst, text, options = nil, powerpack_uuid = nil)
-        valid_param?(:src, src, [Integer, String, Symbol], false)
-        valid_param?(:text, text, [String, Symbol], true)
-        valid_param?(:dst, dst, Array, true)
-        valid_param?(:powerpack_uuid, powerpack_uuid, [String, Symbol], false)
-        dst.each do |dst_num|
-          valid_param?(:dst_num, dst_num, [Integer, String, Symbol], true)
-        end
+      def create(src = nil,  dst = nil, text = nil, options = nil, powerpack_uuid = nil)
 
-        if dst.include? src
-          raise InvalidRequestError, 'src and dst cannot be same'
-        end
+        #All params in One HASH
+        if(src.is_a?(Hash))
+          valid_param?(:src, src[:src], [Integer, String, Symbol], false)
+          valid_param?(:text, src[:text], [String, Symbol], true)
+          valid_param?(:dst, src[:dst], [String, Array], true)
+          valid_param?(:powerpack_uuid, src[:powerpack_uuid], [String, Symbol], false)
 
-        if src.nil? && powerpack_uuid.nil?
-          raise InvalidRequestError, 'src and powerpack uuid both cannot be nil'
-        end
+          if (src[:dst] == src[:src])
+            raise InvalidRequestError, 'src and dst cannot be same'
+          end
 
-        if !src.nil? && !powerpack_uuid.nil?
-          raise InvalidRequestError, 'src and powerpack uuid both cannot be present'
-        end
+          if src.key?(:src).nil? && src.key(:powerpack_uuid).nil?
+            raise InvalidRequestError, 'src and powerpack uuid both cannot be nil'
+          end
 
-        params = {
-          src: src,
-          dst: dst.join('<'),
-          text: text,
-          powerpack_uuid: powerpack_uuid
-        }
+          if !src.key?(:src).nil? && !src.key(:powerpack_uuid).nil?
+            raise InvalidRequestError, 'src and powerpack uuid both cannot be present'
+          end
 
-        return perform_create(params) if options.nil?
-        valid_param?(:options, options, Hash, true)
+          params = {
+            src: src[:src],
+            dst: src[:dst].split(//).join('<'),
+            text: src[:text],
+            powerpack_uuid: src[:powerpack_uuid]
```

Comment:
is this option[:powerpack_uuid] or. src[:powerpack_uuid] ? Please check for other param too?

Reply:
It's src[:powerpack_uuid] and checked for other params too.

Reply:
give valid hash name and mention in param defination

Reply:
Done